pkg/util/graphqlutil/graphiql.go

```diff
@@ -0,0 +1,65 @@
+package graphqlutil
+
+import (
+	htmltemplate "html/template"
+	"net/http"
+)
+
+var graphiqlTemplate = htmltemplate.Must(htmltemplate.New("graphiql").Parse(`<!DOCTYPE html>
+<html lang="en">
+<head>
+	<meta charset="UTF-8">
+	<meta name="viewport" content="width=device-width, initial-scale=1.0">
+	<title>{{ .Title }}</title>
+	<style>
+		body {
+			padding: 0;
+			margin: 0;
+			min-height: 100vh;
+		}
+		#root {
+			height: 100vh;
+		}
+	</style>
+	<script
+		crossorigin
+		src="https://unpkg.com/react@17.0.2/umd/react.production.min.js"
+	></script>
+	<script
+		crossorigin
+		src="https://unpkg.com/react-dom@17.0.2/umd/react-dom.production.min.js"
+	></script>
+	<link rel="stylesheet" href="https://unpkg.com/graphiql@2.0.13/graphiql.min.css" />
+</head>
+<body>
+	<div id="root">Loading...</div>
+	<script
+		crossorigin
+		src="https://unpkg.com/graphiql@2.0.13/graphiql.min.js"
+	></script>
+	<script>
+		ReactDOM.render(
+			React.createElement(GraphiQL, {
+				fetcher: GraphiQL.createFetcher({
+					url: "",
+				}),
+				query: (new URLSearchParams(window.location.search)).get("query") || "",
```

Comment:
Currently, the query function is partly broken. When `query`, `variables`, and `operationName` are provided, only `query` is pre-filled. When I updated the `query` in the UI, the URL query was not updated either. 

I checked the source code of `graphql-go` and found they have some logic to sync the query parameter.
refs: https://github.com/graphql-go/handler/blob/f96ffdde846be75dd40541aebd1dba604f274817/graphiql.go#L21

So I think maybe we should either:
1. Drop query parameter support in the new version? We may need to discuss on this
2. Repeat what `graphql-go` is doing, so the query is synced in 2 ways.

Thought?

Reply:
I tried repeating what `graphql-go` is doing and it seems working well. See if it looks good for you?

https://github.com/carmenlau/authgear-server/commits/2622-graphiql

Reply:
I did not drop it because we need `query` to support a use case in the portal. In the Admin API page in the portal, we have an URL to open GraphiQL with a default query. I did not copy graphql-go was doing because GraphiQL now have tabs, which makes sync query with an arbitrary tab looks weird. So in this sense the meaning of `query` changes, it means initial query now.